cmd/medius/images/promote.go

```diff
@@ -0,0 +1,74 @@
+package images
+
+import (
+	"context"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/repository"
+)
+
+func NewPromoteImagesCommand(options *common.Options) *cobra.Command {
+	options.PromoteImageOptions = common.PromoteImageOptions{
+		TargetRegistry: "quay.io/containerdisks",
+	}
+
+	promoteCmd := &cobra.Command{
+		Use:   "promote",
+		Short: "Promote verified containerdisks from one registry to another registry",
+		Run: func(cmd *cobra.Command, args []string) {
+			results, err := readResultsFile(options.ImagesOptions.ResultsFile)
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			err = spawnWorkers(cmd.Context(), options, func(a api.Artifact) error {
```

Comment:
I think it would be great if we can store in the cache file on every pipeline step if the step failed for a specifc image (like on push, on verify, on promote). Finally we can then fail the `promote` command and print out all images which failed, so that we can investigate them.

Note that this allows us to easily identify if something goes wrong (since the CI job will finally fail), while the service is still operational at the same time and continues publishing.